### DISCUSSION TITLE
Using snapshots like a .lock file

### DISCUSSION THREAD

**Discussion by YodaEmbedding** [wbthomason/packer.nvim#914] [Q&A]
How do I use packer snapshots in the exact same way as a `.lock` file?

[This](https://github.com/YodaEmbedding/dotfiles/blob/3599c4df0ba62bc59dd290c1ea029d161f419c4d/nvim/.config/nvim/lua/plugins/_packer.lua#L132-L139) almost works:

```lua
packer.startup {
  packer_spec,
  config = {
    snapshot = util.join_paths(vim.fn.stdpath("config"), "packer-lock.json"),
    snapshot_path = vim.fn.stdpath("config"),
  },
}
```

...except that it _always_ tries to rollback on every neovim startup, not merely on `:PackerSync`, which is how `.lock` files usually work.

Output of `:messages`:

```
Rollback to "/home/username/dotfiles/nvim/.config/nvim/packer-lock.json" completed
```

Also, annoyingly, I have to press enter every time to hide this status message.

**Comment by allworldg** [comment]
I got same question today.

**Comment by danielo515** [comment]
Sadly, snapshots are not intended for that use case. They are just "pictures" (snapshots if you will 😄 ) of certain states of your plugins, so you can go back to them at any moment.
There is already a proposal to provide and use lockfiles #1009 and even a PR.
You can also provide a commit to the plugin configuration. And you can derive that from a snaphsot if you want by using some functions in your neovim config if you want (I have some utility functions to do that)

**Reply by EdenEast** [reply]
Having more people test the feature would be a great help. I would also suggest that you watch the issue and PR for changes. I have created a feature proposal and implementation but it might change what it looks like when Wil reviews it.

**Reply by allworldg** [reply]
”pictures“ is great , but it is a little annoyed because snapshot try to roll back everytime when neovim open .

**Comment by allworldg** [comment]
@YodaEmbedding Now Packer.nvim update the `SnapShot` way . If you want to create a `SnapShot` , just type the command : `PackerSnapShot v1` . `v1` just a snapshot name .  When you want to rollback , type `PackerSnapShot v1 <plugin Name> `. 

my config:
```
packer.init(
  {
    snapshot_path = require("packer.util").join_paths(vim.fn.stdpath("config"), "snapshot")
  }
)

```
